daemon/cmd/daemon.go

```diff
@@ -326,15 +327,25 @@ func (d *Daemon) restoreCiliumHostIPs(ipv6 bool, fromK8s net.IP) {
 		default:
 			cidrs = []*cidr.CIDR{node.GetIPv4AllocRange()}
 		}
-		fromFS = node.GetInternalIPv4Router()
+		fromFS = node.GetInternalIPv4Router().AsSlice()
 	}
 
-	restoredIP := node.RestoreHostIPs(ipv6, fromK8s, fromFS, cidrs)
-	if err := removeOldRouterState(ipv6, restoredIP); err != nil {
-		log.WithError(err).Warnf(
-			"Failed to remove old router IPs (restored IP: %s) from cilium_host. Manual intervention is required to remove all other old IPs.",
-			restoredIP,
-		)
+	k8sAddrs, k8sOk := ip.AddrFromIP(fromK8s)
+	if !k8sOk {
+		log.Warn("Failed to convert IP address %q", fromK8s)
```

Comment:
Is it ok to ignore it and fall through to the next step? Same for the below one.